sql/core/src/main/scala/org/apache/spark/sql/execution/Exchange.scala

```diff
@@ -210,7 +210,13 @@ private[sql] case class EnsureRequirements(sqlContext: SQLContext) extends Rule[
 
         def addShuffleIfNecessary(child: SparkPlan): SparkPlan = {
           if (!child.outputPartitioning.guarantees(partitioning)) {
-            Exchange(partitioning, child)
+            // SPARK-9563: We should remove repartition operators
```

Comment:
This is a case where I'd go with a much more verbose comment explaining what's happening here so that readers don't need to go to the JIRA to figure out why this was added (maybe copy and shorten some of my description from the JIRA).